Filter items ijoffe

ijoffe opened the pull request:
See my comment on PR #64. This is an edited version of that branch that I think is an improvement, the team can review and merge whichever they prefer

sumaiyasalsabil commented:
This looks really good, it definitely improved it. I'm okay with merging this one if David is.

ijoffe commented:
should be good to merge. updates i made from david and sumaiya's version:
* some simple refactoring to make the code easier to read imo
* made it so filtering date popups cannot select future dates
* added to javadocs for new stuff
* removed unnecessary fragment (should be simpler and better imo)
* fixed estimated value formatting inconsistency

sumaiyasalsabil reviewed (approved):
looks good
